sub-packages/bionemo-evo2/src/bionemo/evo2/run/train.py

```diff
@@ -550,47 +551,80 @@ def train(args: argparse.Namespace):
-                f"-PEOD{args.eod_pad_in_loss_mask}"
-                f"-BO{args.add_bias_output}"
-                f"-GCLP{args.clip_grad}"
-                f"-HDO{args.hidden_dropout}"
-                f"-ADO{args.attention_dropout}"
-                f"-LR{args.lr}-MINLR{args.min_lr}-WUSTEPS{args.warmup_steps}-WD{args.wd}"
-                f"-GRFP32{args.grad_reduce_in_fp32}-FP8WG{args.fp8_wgrad and args.fp8}"
-                f"-OGR{args.overlap_grad_reduce}-OPG{args.overlap_param_gather}"
-                f"-NODES{args.num_nodes}-FP8{args.fp8}"
-            ),
+    wandb_run_name = (
+        f"evo2-size-{args.model_size}-TP{args.tensor_parallel_size}-"
+        f"PP{args.pipeline_model_parallel_size}-CP{args.context_parallel_size}"
+        f"-GBS{global_batch_size}-MBS{args.micro_batch_size}-SkipLossRenorm{args.no_renormalize_loss}"
+        f"-NOAC{args.no_activation_checkpointing}-SELAC{args.selective_activation_checkpointing}"
+        f"-ACRNL{evo2_config.recompute_num_layers}"
+        f"-PAT{evo2_config.hybrid_override_pattern}"
+        f"-F32R{evo2_config.fp32_residual_connection}"
+        f"-FCE{evo2_config.cross_entropy_loss_fusion}"
+        f"-AIC{not args.no_average_in_collective}"
+        f"-PEOD{args.eod_pad_in_loss_mask}"
+        f"-BO{args.add_bias_output}"
+        f"-GCLP{args.clip_grad}"
+        f"-HDO{args.hidden_dropout}"
+        f"-ADO{args.attention_dropout}"
+        f"-LR{args.lr}-MINLR{args.min_lr}-WUSTEPS{args.warmup_steps}-WD{args.wd}"
+        f"-GRFP32{args.grad_reduce_in_fp32}-FP8WG{args.fp8_wgrad and args.fp8}"
+        f"-OGR{args.overlap_grad_reduce}-OPG{args.overlap_param_gather}"
+        f"-NODES{args.num_nodes}-FP8{args.fp8}"
+    )
+
+    wandb_config: Optional[WandbConfig] = (
+        None
+        if args.wandb_project is None
+        else WandbConfig(
+            offline=args.wandb_offline,
+            project=args.wandb_project,
+            name=wandb_run_name,
+            entity=args.wandb_entity,
+            tags=args.wandb_tags,
             group=args.wandb_group,
             job_type=args.wandb_job_type,
-            id=args.wandb_run_id,
-            project=args.wandb_project,
-            save_dir=args.experiment_dir,
-            offline=args.wandb_offline,
+            id=args.wandb_id,
             anonymous=args.wandb_anonymous,
+            log_model=args.wandb_log_model,
         )
-        loggers.append(wandb_logger)
-        nemo_logger_kwargs["wandb"] = wandb_logger
-    tb_logger = TensorBoardLogger(
-        save_dir="dummy",  ## NOTE: this gets overwritten by default
     )
-    nemo_logger_kwargs["tensorboard"] = tb_logger
-    loggers.append(tb_logger)
+    nemo_logger = setup_nemo_lightning_logger(
+        root_dir=args.result_dir,
+        name=args.experiment_name,
+        initialize_tensorboard_logger=args.create_tensorboard_logger,
+        wandb_config=wandb_config,
+    )
+
+    if args.create_checkpoint_callback:
```

Comment:
put together model checkpoint and autoresume since both require the same checkpoint_dirpath and are needed only if model checkpoints is created